questionpy_sdk/webserver/static/script.js

```diff
@@ -0,0 +1,182 @@
+import * as conditions from "./conditions.js";
+
+
+/**
+ * Selects all elements which have a list of conditions as a data attribute. The lists of conditions are an
+ * unparsed JSON string and represent either a hide_if list or a disable_if list (see: {@link conditions.Types}).
+ *
+ *
+ * @returns {[Element]} list of elements
+ */
+function get_elements_with_conditions() {
+    let elements_with_conditions = []
+    for (const value in conditions.Types.values) {
+        const condition_type = new conditions.Types(value);
+        document.querySelectorAll(condition_type.to_selector())
+            .forEach(element => {
+                add_conditions_to_element(element, condition_type)
+                elements_with_conditions.push(element);
+            });
+    }
+    return elements_with_conditions;
+}
+
+
+/**
+ * Parses the condition list from the data attribute of the element into a list of {@link conditions}.
+ *
+ * @param {Element} element a form element
+ * @param {object} element.conditions object containing the list of conditions
+ * @param {conditions.Types} condition_type the type of the condition list
+ * @param {string} condition_type.value from ['disable_if', 'hide_if']
+ * @return void
+ */
+function add_conditions_to_element(element, condition_type){
+    if (!element.conditions) {
+        element.conditions = {};
+    }
+    element.conditions[condition_type.value] = [];
+    JSON.parse(element.getAttribute("data-"+condition_type.value)).forEach(object => {
+        let condition = conditions.Condition.from_object(object);
+        add_source_element_to_target(condition.target, element, condition_type);
+        element.conditions[condition_type.value].push(condition);
+    });
+}
+
+
+/**
+ * Adds the source element to the targets list of source elements.
+ * Adds the event listener corresponding to the condition_type to the target.
+ * When the targets event listener is triggered, the source list contains all the elements which conditions have to be
+ * checked.
+ *
+ * @param {HTMLElement} target the conditions target
+ * @param {Element} source element with the condition
+ * @param {conditions.Types} condition_type
+ * @param {string} condition_type.value from ['disable_if', 'hide_if']
+ */
+function add_source_element_to_target(target, source, condition_type) {
+    if (!target['source_elements']) {
+        target['source_elements'] = []
+    }
+    target['source_elements'].push(source);
```

Comment:
```suggestion
    if (!target.source_elements) {
        target.source_elements = []
    }
    target.source_elements.push(source);
```

`a["b"]` und `a.b` sind in JS gleichbedeutend

Reply:
Meine IDE hatte gemeckert wegen `unresolved variable`s, deswegen hatte ich zur array Notation gewechselt.

Reply:
Nervig, aber okay. So passt es auch